app/views/layouts/_hero.html.slim

```diff
@@ -0,0 +1,21 @@
+- if content_for? :hero
+  div class='dfe-content-page--header govuk-!-padding-bottom-4'
+    .dfe-width-container
```

Comment:
Could we flip it and thereby prevent duplication by having a partial for everything inside `.dfe-width-container` and move the `if/else` into the layout.

Reply:
I don't think so, not without having two, almost identical partials, one that yields the hero content and one that doesn't and then having the if/then in the layout.  this looks cleaner to me.

Reply:
How do you feel about a second partial then, used inside this one?

Reply:
I'd say premature DRYing of incidental duplication and not necessary.

Reply:
As an experiment, here is in a more DRY fashion.  Makes for unreadable code.

```
- content_for :banner_timeout_and_backlink do 
  = govuk_phase_banner(tag: { text: 'Beta' }, classes: 'noprint') do
    == t 'phase_banner', link: govuk_link_to('feedback', Rails.configuration.feedback_url, target: :_blank)

  .noprint = render 'layouts/timeout'

  - if request.path != '/email-preferences'
    .noprint = back_link

- if content_for? :hero
  div class='dfe-content-page--header govuk-!-padding-bottom-4'
    .dfe-width-container
      = yield :banner_timeout_and_backlink
      = yield :hero
- else 
  .dfe-width-container
    = yield :banner_timeout_and_backlink
```